content/3_Silver/Binary_Search.mdx

```diff
@@ -414,95 +403,36 @@ int lastTrue(function<bool(int)> f, int lo, int hi) {
 
 <JavaSection>
 
-We can shorten the function by removing `res` and using a ternary if expression.
-
 ```java
-public static int lastTrue(int lo, int hi) {
-	for (lo --; lo < hi; ) {
-		int mid = (lo+hi+1)/2;
-		if(f(mid)) lo = mid; else hi = mid-1;
-	}
+public static int lastTrue(int lo, int hi, Predicate<Integer> f) {
+	for (int dif = (hi-(--lo)); dif>0; dif /= 2)
+		while (lo+dif <= hi && f.test(lo+dif)) lo += dif; //f is the function
 	return lo;
 }
 ```
 
 </JavaSection>
 
-<PySection> -->
-
-<!--
-*Note: it is NOT recommended to use tenary expression in the python implementation of binary search because python tenary expression does not accept statements.*
+<PySection>
 
-<Spoiler title="Python ternary expression">
+<Warning>
+It is NOT recommended to use tenary expression in the python implementation of binary search because python tenary expression does not accept statements.
```

Comment:
```suggestion
It is NOT recommended to use ternary expressions in the Python implementation of binary search because Python ternary expressions do not accept statements.
```